### PR TITLE
add uart_getchar, and flush stdout/stderr in sim script

### PR DESCRIPTION
**Pull request by fuzzie**

### CODE REVIEW

**Comment by knz** [comment]
rebase?

**Comment by fuzzie** [comment]
Rebased.

File: slc/tools/leon2mt-fpga-ctl.in [hunk 1]
```diff
@@ -130,7 +130,9 @@ while True:
 
     output = splitBuffer(mybuf)
     sys.stdout.write(output[0])
+    sys.stdout.flush()
```
Comment (knz) [inline slc/tools/leon2mt-fpga-ctl.in:133]: why the extra flush?

**Review by knz — COMMENTED** [review]
?

File: slc/tools/leon2mt-fpga-ctl.in (hunk 1 above)
Comment (fuzzie) [reply slc/tools/leon2mt-fpga-ctl.in:133]: To make sure output is displayed immediately; by default I'm only seeing flushes on newlines, which is pretty confusing while debugging.

**Comment by knz** [comment]
ok thanks

**Pull request merged**